File: fixtures/lang-coverage/haskell/src/Service.hs
module Service where

-- feature: module, import (plain, qualified, selective), function, call

import Types
import Helpers (formatName, unrelatedHelper)
import qualified Data.Map as M
import Data.Maybe (fromMaybe)

maxRetries :: Int
maxRetries = 3

runService :: Service -> IO ()
runService svc = do
  let msg = formatName (serviceName svc)
  putStrLn msg
  let _ = unrelatedHelper maxRetries
  return ()

makeService :: Id -> Name -> Service
makeService i n = Service { serviceId = i, serviceName = n }

lookupState :: M.Map String State -> String -> State
lookupState m k = fromMaybe Idle (M.lookup k m)

File: fixtures/lang-coverage/julia/src/Service.jl
module Service

# feature: module, using, import, function, struct, const, call

using LinearAlgebra
import Helpers: format_name

export run_service, make_service

const SERVICE_VERSION = "1.0"

struct ServiceConfig
    name::String
    max_retries::Int
end

function make_service(name::String)
    return ServiceConfig(name, 3)
end

function run_service(cfg::ServiceConfig)
    msg = format_name(cfg.name)
    println(msg)
    return cfg.max_retries
end

end # module
